views/NewChart.react.js

```diff
@@ -1,9 +1,107 @@
+
+import getCounts from './fe-utils/countUtil'
+import getAvatar from './fe-utils/avatarUtil'
+import { fetchSubjects, createChart } from './fe-utils/fetchUtil'
+import getDefaultStyles from './fe-utils/styleUtil'
+import { chartStyles } from './styles/chart_styles'
+
+const NewChart = ({ user, classes }) => {
+  const [formValues, setFormValues] = useState({})
+  const [fieldLabelValueMap, setFieldLabelValueMap] = useState([])
+  const [openDrawer, setOpenDrawer] = useState(false)
+  const [sideBarState, setSideBarState] = useState({ totalDays: 0, totalStudies: 0, totalSubjects: 0 })
+  const [avatar, setAvatar] = useState('')
+
+  const toggleDrawer = () => setOpenDrawer(!openDrawer)
+  const updateFormValues = (e,) => setFormValues({...formValues, [e.target.name]: e.target.value})
+  const addValueAndLabelField = () => setFieldLabelValueMap(prevState => [{ value: '', label: '' }, ...prevState ])
+  const removeValueAndLabelField = (id) => setFieldLabelValueMap(prevState => [...prevState.filter((_, index) => index !== id)])
+  const handleValueAndLabelFieldUpdate = (e, id) => setFieldLabelValueMap(prevState => 
+    [...prevState
+      .map((field, idx) => 
+        id === idx
+        ? 
+          ({ ...field, [e.target.name]: e.target.value }) 
+        : 
+          ({...field})
+        )
+      ])
+  const handleSubmit = async (e) => {
+    e.preventDefault()
+    const values = { ...formValues, fieldLabelValueMap: fieldLabelValueMap }
+    await createChart(values)
+  }
+
+  useEffect(() => {
+    fetchSubjects().then(acl => {
+      setSideBarState(getCounts({ acl }))
+    })
+    setAvatar(getAvatar({ user: user }))
+  }, [])
 
-export default function NewChart() {
   return (
-    <div>
-      <h1> New Chart</h1>
+    <div className={ classes.root }>
+      <Header
+        handleDrawerToggle={toggleDrawer}
+        title='Create a New Chart'
+        isAccountPage={false}
+      />
+      <Sidebar
+        avatar={avatar}
+        handleDrawerToggle={toggleDrawer}
+        mobileOpen={openDrawer}
+        totalDays={sideBarState.totalDays}
+        totalStudies={sideBarState.totalStudies}
+        totalSubjects={sideBarState.totalSubjects}
+        user={user}
+      />
+      <>
+        <div className={`${ classes.content } ${ classes.contentPadded }`}>
+          <Form handleSubmit={handleSubmit}>
+            <BarChartFields 
+              classes={classes} 
+              updateFormValues={updateFormValues} 
+              formValues={formValues}
+              fieldLabelValueMap={fieldLabelValueMap}
+              addValueAndLabelField={addValueAndLabelField}
+              removeField={removeValueAndLabelField}
+              updateFieldValues={handleValueAndLabelFieldUpdate}
+            />
+            <div className={classes.submitButtonContainer}>
+              <Button
+                type='submit'
+                variant='raised'
+                className={classes.textButton}
+                onClick={handleSubmit}
```

Comment:
do you need this click handler if the form itself has a `handleSubmit` prop?

Reply:
[8a7c3d0](https://github.com/AMP-SCZ/dpdash/pull/84/commits/8a7c3d014d0266427e76ae9b5d15d0618767ac3a)